From 982b524bb8da169414b4518f3a2f961862f7a34b Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Fri, 6 Jan 2017 10:01:19 +0800
Subject: [PATCH 1384/5242] MLK-13720-6 ARM: imx: mu: remove
 clk_prepare_enable in resume

commit  28ae0e85ab15f332aebdab5a0e4327b32ccc1022 from
https://source.codeaurora.org/external/imx/linux-imx.git

We didn't need to touch MU clock on i.mx7ULP, since it has already
been enabled by M4 side, remove the useless code in resume flow

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
(cherry picked from commit 117f17a00b2508c783c13dea090a2896689b5601)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/mu.c |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/arch/arm/mach-imx/mu.c b/arch/arm/mach-imx/mu.c
index a600054..c2cca7e 100644
--- a/arch/arm/mach-imx/mu.c
+++ b/arch/arm/mach-imx/mu.c
@@ -491,17 +491,9 @@ static int mu_suspend(struct device *dev)
 
 static int mu_resume(struct device *dev)
 {
-	int ret;
-
 	if (!cpu_is_imx7ulp())
 		return 0;
 
-	ret = clk_prepare_enable(clk);
-	if (ret) {
-		dev_err(dev, "unable to enable mu clock\n");
-		return ret;
-	}
-
 	writel_relaxed(readl_relaxed(mu_base + MX7ULP_MU_CR) |
 			BIT(0) | BIT(26) | BIT(27), mu_base + MX7ULP_MU_CR);
 
-- 
1.7.9.5

